engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -0,0 +1,204 @@
+		localPart = localPart.replace( "`", "");
+		localPart = localPart.replace( "{", "");
+		localPart = localPart.replace( "|", "");
+		localPart = localPart.replace( "}", "");
+		localPart = localPart.replace( "~", "");
+		localPart = localPart.replace( ".", "");
+		Assertions.assertThat(localPart).isEqualTo("");
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void validUseOfDotBeforeAt(@ForAll("emails") String email){
+		String localPart = getLocalPartOfEmail(email);
+		Assume.that(!isQuoted(localPart));
+		Assertions.assertThat(localPart).doesNotContain("..");
+		Assertions.assertThat(localPart.charAt(0)).isNotEqualTo('.');
+		Assertions.assertThat(localPart.charAt(localPart.length() - 1)).isNotEqualTo('.');
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void validSignsAfterAt(@ForAll("emails") String email){
+		String domain = getDomainOfEmail(email);
+		Assume.that(!isIPAddress(domain));
+		for(char c = 'a'; c <= 'z'; c++){
+			domain = domain.replace(c + "", "");
+		}
+		for(char c = 'A'; c <= 'Z'; c++){
+			domain = domain.replace(c + "", "");
+		}
+		for(char c = '0'; c <= '9'; c++){
+			domain = domain.replace(c + "", "");
+		}
+		domain = domain.replace("-", "");
+		domain = domain.replace(".", "");
+		Assertions.assertThat(domain).isEqualTo("");
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void validUseOfHyphenAndDotAfterAt(@ForAll("emails") String email){
+		String domain = getDomainOfEmail(email);
+		Assume.that(!isIPAddress(domain));
+		Assertions.assertThat(domain.charAt(0)).isNotEqualTo('-');
+		Assertions.assertThat(domain.charAt(domain.length() - 1)).isNotEqualTo('-');
+		Assertions.assertThat(domain).doesNotContain("..");
+		Assertions.assertThat(domain.charAt(0)).isNotEqualTo('.');
+		Assertions.assertThat(domain.charAt(domain.length() - 1)).isNotEqualTo('.');
+		Assume.that(domain.length() >= 2);
+		Assertions.assertThat(domain.charAt(domain.length() - 2)).isNotEqualTo('.');
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void validMaxDomainLengthAfterAt(@ForAll("emails") String email){
+		String domain = getDomainOfEmail(email);
+		Assume.that(!isIPAddress(domain));
+		String[] domainParts = domain.split("\\.");
+		IntStream.range(0, domainParts.length).forEach(i -> {
+			Assertions.assertThat(domainParts[i].length()).isLessThanOrEqualTo(63);
+		});
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void validIPAddressAfterAt(@ForAll("emails") String email){
+		String domain = getDomainOfEmail(email);
+		Assume.that(isIPAddress(domain));
+		domain = domain.substring(1, domain.length() - 1);
+		Assertions.assertThat(InetAddresses.isInetAddress(domain)).isTrue();
+	}
+
+	@Property(edgeCases = EdgeCasesMode.NONE)
+	void quotedAndUnquotedUsernamesAreGenerated(@ForAll("emails") String email){
+		String localPart = getLocalPartOfEmail(email);
+		Statistics.collect(isQuoted(localPart));
+		Statistics.coverage(coverage -> {
+			coverage.check(true).count(c -> c >= 1);
+			coverage.check(false).count(c -> c >= 1);
```

Comment:
Maybe percentage checks `.percentage(p -> p > 40%)` describe better what you want?